openwisp_notifications/base/models.py

```diff
@@ -51,12 +55,19 @@ def message(self):
             )
 
             config = get_notification_configuration(self.type)
-            if 'message' in config:
-                md_text = config['message'].format(notification=self)
-            else:
-                md_text = render_to_string(
-                    config['message_template'], context=dict(notification=self)
-                ).strip()
+            try:
+                if 'message' in config:
+                    md_text = config['message'].format(notification=self)
+                else:
+                    md_text = render_to_string(
+                        config['message_template'], context=dict(notification=self)
+                    ).strip()
+            except AttributeError as e:
+                logger.error(e)
+                md_text = (
+                    'Attribute Error while generating notification message.'
+                    'Check logs for further information.'
+                )
```

Comment:
I don't know, this feels relaly wrong to me.

The best thing would be to skip the notification entirely, but I guess we cannot do that in the admin.

However, I believe we can do this once we use the JS widget to show notifications.

I think the best course of action is the following. For now, just make the message more generic: 
`Error while generating notification message, notification data may have been deleted`.
I don't think we should advise users to check logs, because it's entirely possible they may not have access to the application logs.

Later on, once we have the notification widget and we stop using the django admin, we can raise an exception (a custom one we create for this purpose), catching this exception in the API and websocket when iterating on the notification list, in case the exception is catched, the rendering of the notification is just skipped.

Will create an issue for this.